Add step-61, a program that uses the weak Galerkin method

sophy1029 opened the pull request.

bangerth commented:
By the way, if you have time between 2 and 3pm, please come by my office and we can chat about next steps.

sophy1029 commented:
Yes, I will be there around 2:30pm. I need to go to the measure theory
class to get my take home exam first. Thanks.

On Fri, May 4, 2018 at 11:25 AM, Wolfgang Bangerth <notifications@github.com
> wrote:

> By the way, if you have time between 2 and 3pm, please come by my office
> and we can chat about next steps.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bangerth/dealii/pull/6#issuecomment-386670355>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AdMXNwpEVwPS-nDVuRsisO9PzDhn-ue-ks5tvI77gaJpZM4Tqrsh>
> .
>

Commit:
add step-63.cc (no doc)

bangerth committed:
Edit the introduction of step-61.

jppelteret committed:
Add SymEngine macros to config.h

bangerth committed:
Allow initialization of Vector with std::initializer_list.

bangerth committed:
Add changelog.

bangerth committed:
Add a test.

bangerth committed:
Fix a changelog entry.

masterleinad committed:
Merge pull request #7864 from jppelteret/rename_solver_class

Rename Solver (base) class to SolverBase

masterleinad committed:
Merge pull request #7872 from bangerth/fix-changelog-entry

Fix a changelog entry.

jppelteret committed:
Merge pull request #7871 from bangerth/initializer_list

Allow initialization of Vector with std::initializer_list.

jppelteret committed:
Add traits for SD numbers

jppelteret committed:
Add basis for SymEngine (SD) wrappers

- Types
- Expression class
- Utilities

jppelteret committed:
Add test for SD::Expression class

bangerth committed:
Use a range-based for loop.

drwells committed:
Fix a comment.

There are no blocks in this object.

drwells committed:
Clean up some headers and code samples.

drwells committed:
Remove '__' from include guards.

tjhei committed:
CI: archive detailed.log

drwells committed:
Merge pull request #7874 from drwells/underscore-guards

Remove '__' from include guards.

drwells committed:
Merge pull request #7873 from drwells/vector-header

Clean up the vector header a bit.

jppelteret committed:
Add functions to perform symbolic math operations using SD::Expression

jppelteret committed:
Add test for symbolic math functions

masterleinad committed:
Update PETSc version compatibility

tjhei committed:
fix path

drwells committed:
Merge pull request #7877 from masterleinad/update_petsc_version_number

Update PETSc version compatibility

drwells committed:
Merge pull request #7878 from bangerth/for

Use a range-based for loop in a couple of places.

jppelteret committed:
Merge pull request #7875 from tjhei/jenkins_archive_detailedlog

CI: archive detailed.log

tjhei committed:
several doxygen errors and warnings

jppelteret committed:
Add a class to perform GeneralDataStorage

masterleinad committed:
Unify intrinsics headers

kronbichler committed:
Merge pull request #7879 from tjhei/doxygen_fixes2

several doxygen errors and warnings

bangerth committed:
Merge pull request #7881 from masterleinad/unify_intrinsic_headers

Unify intrinsics headers

davydden committed:
set LAPACK_INCLUDE_DIRS and LAPACK_USER_INCLUDE_DIRS

tamiko committed:
Merge pull request #7884 from davydden/config/lapack_include

set LAPACK_INCLUDE_DIRS and LAPACK_USER_INCLUDE_DIRS

davydden committed:
add default constructor for ArrayView and a reinit() method

masterleinad committed:
Merge pull request #7885 from davydden/feature/array_view_default

add default constructor for ArrayView and a reinit() method

masterleinad committed:
Test SCALAPACK with -Werror

davydden committed:
Merge pull request #7886 from masterleinad/scalapck_werror

Test SCALAPACK with -Werror

tjhei committed:
fix hdf in serial

- fix compilation
- fix warnings about unused variables
- move some variables around (smaller scope)

masterleinad committed:
Fix space in error message

jppelteret committed:
Merge pull request #7890 from masterleinad/fix_space

Fix space in error message

jppelteret committed:
Fix a typo in a changelog entry

drwells committed:
Merge pull request #7746 from bangerth/61

More edits in the introduction of step-61.

bangerth committed:
Fix a misspelled word.

kronbichler committed:
Merge pull request #7893 from bangerth/spell

Fix a misspelled word.

kronbichler committed:
Merge pull request #7889 from tjhei/hdf_no_mpi

fix hdf5 in serial

luca-heltai committed:
Merge pull request #7844 from jppelteret/sd-expression

SD: Types, traits, and expression class

luca-heltai committed:
Fixed documentation.

drwells committed:
Silence an unused variable warning.

Commit:
requested changes

drwells committed:
Merge pull request #7894 from drwells/unused-var-hdf5

Silence an unused variable warning.

jppelteret committed:
Add alternative test output

jppelteret committed:
Try to reproduce GCC tester error with ambiguous constructor

masterleinad committed:
Require SymEngine version 0.4

tjhei committed:
mpi quicktests fail

mpirun completes about not finding the executable. This reverts part of

tjhei committed:
[CI]: run quicktests

also execute the quicktests as part of CI

tjhei committed:
CI: catch failures

masterleinad committed:
Fix ambiguous function call

kkremitzki committed:
Update dead LICENSE link in README

masterleinad committed:
Merge pull request #7900 from kkremitzki/fix-README

Update dead LICENSE link in README

masterleinad committed:
Use std::string in SmartPointer and Subscriptor interface

tamiko committed:
Merge pull request #7896 from masterleinad/update_minimum_symengine_version

Require SymEngine version 0.4

masterleinad committed:
Merge pull request #7897 from tjhei/fix_mpi_quicktests

MPI quicktests failing

masterleinad committed:
Replace empty string comparison

masterleinad committed:
Add overload for default constructors

masterleinad committed:
Use references and clean up a bit

luca-heltai committed:
Merge pull request #7880 from jppelteret/general_data_storage

Add a class to perform GeneralDataStorage

luca-heltai committed:
CopyData

luca-heltai committed:
ScratchData

luca-heltai committed:
Test ScratchData on simple volume computation

luca-heltai committed:
Test ScratchData for Laplacian (conforming)

luca-heltai committed:
Test ScratchData for Laplacian (SIPDG)

luca-heltai committed:
Test ScratchData get_values

luca-heltai committed:
ScratchData with Sacado applied to Residual

luca-heltai committed:
ScratchData + Sacado applied to energies

masterleinad committed:
Merge pull request #7898 from tjhei/ci_quicktests

[CI]: catch failures and run quicktests

luca-heltai committed:
ScratchData + Sacado FD+RD applied to energies

luca-heltai committed:
ChangeLog

rezarastak committed:
Fixed documentation error in filtered_iterator.h

jppelteret committed:
Merge pull request #7906 from rezarastak/filtered_iterator_doc_fix

Fixed documentation error in filtered_iterator.h

luca-heltai committed:
Comments.

luca-heltai committed:
defgroup -> name

luca-heltai committed:
Merge pull request #7806 from luca-heltai/scratch_and_copy

"Reasonable" ScratchData and CopyData

masterleinad committed:
Replace std::make_unique by std_cxx14::make_unique

luca-heltai committed:
Enlarge BoundingBox

luca-heltai committed:
Make sure rtree works with mappings that do not preserve vertices.

luca-heltai committed:
Merge pull request #7910 from masterleinad/replace_std_make_unique

Replace std::make_unique by std_cxx14::make_unique

masterleinad committed:
Merge pull request #7901 from masterleinad/smart_pointer_std_string

Use std::string in SmartPointer and Subscriptor interface

tjhei committed:
clang-tidy: only debug mode

I don't think it makes sense to also compile in release mode for the
tidy checks. This change should speed up the CI tester as well.

luca-heltai committed:
Merge pull request #7916 from tjhei/tidy_debug_only

clang-tidy: only debug mode

masterleinad committed:
Avoid floating point exception in symengine_number_type_01

masterleinad committed:
Merge pull request #7876 from jppelteret/sd-expression-math

SD: Logical operators + math operations for expression class

tjhei committed:
indent

tjhei committed:
get rid of static smoother data

tjhei committed:
change parse_parameter logic

tjhei committed:
edit intro

drwells committed:
Add missing '@ingroup functions' annotations.

jppelteret committed:
Merge pull request #7919 from drwells/add-missing-ingroup-functions

Add missing '@ingroup functions' annotations.

jppelteret committed:
Merge pull request #7917 from masterleinad/avoid_fpe

Avoid floating point exception in symengine_number_type_01

jppelteret committed:
Add some SD-related utility functions

jppelteret committed:
Add some SD-related convenience functions for scalar operations.

Specifically, these can be used to create a scalar symbol and symbolic
function.

jppelteret committed:
Add some SD-related convenience functions for scalar operations.

Specifically, a non-member function for scalar differentiation of an
Expression has been added.

luca-heltai committed:
FunctionFromFunctionObjects.

bangerth committed:
Enclose everything in step-61 into its own namespace.

bangerth committed:
Update requirements for step-61.

masterleinad committed:
Merge pull request #7924 from bangerth/61-2

Update requirements for step-61.

masterleinad committed:
Fix symengine_number_type_01 in Release mode

Commit:
requested changes

tamiko committed:
add a test

Rombur committed:
Remove unused variables

Rombur committed:
Add missing instantiations

luca-heltai committed:
Merge pull request #7925 from masterleinad/avoid_fpe

Fix symengine_number_type_01 in Release mode

luca-heltai committed:
Merge pull request #7923 from bangerth/61

Enclose everything in step-61 into its own namespace.

masterleinad committed:
Merge pull request #7927 from Rombur/cuda_fix

Cuda: small fixes

masterleinad committed:
Merge pull request #7915 from luca-heltai/rtree-with-mapped-vertices

Make sure rtree works with mappings that do not preserve vertices.

masterleinad committed:
Merge pull request #7914 from luca-heltai/enlarge-bounding-box

BoundingBox::extend()

jppelteret committed:
Add template arguments for symmetric tensor ranks

jppelteret committed:
Add ProductType for SD::Expression class.

This is the first step towards support of SD::Expression within the
Tensor and SymmetricTensor classes.

jppelteret committed:
Specialize some functions in the numbers namespace for SD::Expression.

Also required for support of SD::Expression within the Tensor and
SymmetricTensor classes.

jppelteret committed:
Add some SD-related convenience functions for tensor operations.

Specifically, these can be used to create a tensor symbol and symbolic
function.

luca-heltai committed:
Modify FunctionCutOff to support integration to one.

luca-heltai committed:
Deprecate new_* in favour of set_*

jppelteret committed:
Add a test to check std::copy and std::swap with vector of Expressions

luca-heltai committed:
Added FunctionCutOffTensorProduct.

masterleinad committed:
Fix configuring with external Boost-1.70.0

masterleinad committed:
Add links

jppelteret committed:
Add test for operations using symbolic and expressional tensors

luca-heltai committed:
Added implementation of CutOffFunctionC1

BenBrands committed:
add explicit instantiations for shared::Triangulation

isuruf committed:
CMake: Clear CMAKE_REQUIRED_FLAGS before calling FindLAPACK

CMAKE_REQUIRED_FLAGS is populated by dealii with CXXFLAGS and this
should be cleared before calling any function that calls try_compile
on a C file. FindLAPACK calls FindTHREADS.cmake which uses try_compile
on a C file.

tjhei committed:
Merge pull request #7935 from masterleinad/fix_boost_1_70

Fix configuring with external Boost-1.70.0

masterleinad committed:
Merge pull request #7938 from isuruf/cmake

CMake: Clear CMAKE_REQUIRED_FLAGS before calling FindLAPACK

luca-heltai committed:
Mapping::get_center()

bangerth committed:
Mark two function references better.

jppelteret committed:
Add functions to perform SD tensor differentation

jppelteret committed:
Alter the wording describing the input parameter and output of a function

jppelteret committed:
Add two changelog entries

jppelteret committed:
AD: Add test for 4 component scalar function

jppelteret committed:
Add a test for SD tensor differentiation

jppelteret committed:
Merge pull request #7920 from jppelteret/sd-scalar_operations_01

SD scalar operations: Symbol and symbolic function creation

bangerth committed:
Clarify a piece of documentation.

tjhei committed:
Merge pull request #7942 from bangerth/doc

Mark two function references better.

drwells committed:
Merge pull request #7945 from bangerth/doc-2

Clarify a piece of documentation.

tamiko committed:
SparsityPattern: Fix copy from DynamicSparsityPattern

tamiko committed:
avoid a segmentation fault

jppelteret committed:
Merge pull request #7936 from BenBrands/instantiations_shared_triangulation

add explicit instantiations for shared::Triangulation

tamiko committed:
improve error message

luca-heltai committed:
Fix documentation of extend + spelling.

luca-heltai committed:
Improved documentation.

masterleinad committed:
Merge pull request #7931 from jppelteret/sd-tensor_operations_01

SD tensor operations: Symbol and symbolic function creation

bangerth committed:
Merge pull request #7926 from tamiko/fix_dynamic_sparsity_pattern_exists

SparsityPattern: Fix copy_from a DynamicSparsityPattern

masterleinad committed:
Fix SymEngine test

Rombur committed:
Reduce chunk_size used in CUDA kernels from 8 to 1

rezarastak committed:
Documentation fix for DerivativeForm

masterleinad committed:
Merge pull request #7947 from luca-heltai/fix-spelling

Fix documentation of extend + spelling.

bangerth committed:
Merge pull request #7949 from rezarastak/patch-1

Documentation fix for DerivativeForm

bangerth committed:
Merge pull request #7948 from masterleinad/fix_symengine_test

Fix SymEngine test

bangerth committed:
Avoid latex formulas in section headings.

bangerth committed:
Fix a grammar mistake in a comment.

bangerth committed:
Remove pictures from the repository.

Instead, they are now stored in the repo of the website, where they
are also referenced in the program's documentation itself. In
other words, they are no longer necessary in the deal.II git repo
itself.

tjhei committed:
Merge pull request #7951 from bangerth/grammar

Fix a grammar mistake in a comment.

tjhei committed:
Merge pull request #7950 from bangerth/61

Avoid latex formulas in section headings.

bangerth committed:
Break long lines in step-61/doc/results.dox.

tjhei committed:
Merge pull request #7953 from bangerth/61-3

Break long lines in step-61/doc/results.dox.

bangerth committed:
Edit the results section of step-61.

tjhei committed:
step-40 disable logic with complex PETSc

Having PETSC_WITH_COMPLEX would stop step-40 from being able to
configure/build. Now we fall back to trilinos as described in the error
message.

part of #7946

rezarastak committed:
Mathematical explanation for the functions within DerivativeForm is improved

luca-heltai committed:
Added static_assert.

luca-heltai committed:
Merge pull request #7944 from jppelteret/sd-tensor_operations_02

SD tensor operations: Add functions to perform symbolic tensor differentation

kronbichler committed:
Merge pull request #7955 from Rombur/cuda_size

Reduce chunk_size used in CUDA kernels from 8 to 1

drwells committed:
Merge pull request #7956 from tjhei/step-40-cmake-complex

step-40 disable logic with complex PETSc

jppelteret committed:
Add functions to create SD subsitution maps from scalar types

bangerth committed:
Apply suggestions from code review

Co-Authored-By: rezarastak <rezarastak@users.noreply.github.com>

bangerth committed:
Augment documentation in class IteratorRange.

kronbichler committed:
Documentation fix in Physics::Transformations::Rotations::rotation_matrix_2d

bangerth committed:
Fix a mistaken assertion that only succeeds by accident.

Specifically, it succeeds because we get a pointer to the first of a block of
elements where the size of the block is greater than one. If one had a 1-point
quadrature formula, the assertion would fail.

bangerth committed:
Merge pull request #7932 from luca-heltai/dirac-delta-cutoff

Enrich CutOffFunctionBase, and provide also CutOffFunctionC1

rezarastak committed:
asterisk removed from notation in DerivativeForm

bangerth committed:
Merge pull request #7918 from luca-heltai/lambda-function

FunctionFromFunctionObjects class: wraps a vector of std::function objects into a Function object

jppelteret committed:
Merge pull request #7959 from kronbichler/doc_rotation_matrix_2d

Documentation fix Physics::Transformations::Rotations::rotation_matrix_2d

jppelteret committed:
Implement function to merge substitution maps

jppelteret committed:
Add test for addition of scalar types to SD substitution maps

rezarastak committed:
Documentation added so that the using statements are properly shown in the manual

jppelteret committed:
Add test for merging of SD substitution maps

rezarastak committed:
small fix

luca-heltai committed:
Merge pull request #7960 from bangerth/fix-18

Fix a mistaken assertion that only succeeds by accident.

bangerth committed:
Merge pull request #7962 from rezarastak/generic_la_documentation

Adding documentation for generic LA namespaces

Commit:
Made degree of quadrature depending on the degree of the finite element used in the program

rezarastak committed:
small documentation fix: replaced (*this) with the matrix A

masterleinad committed:
Deal with empty sizes in CUDA kernels

masterleinad committed:
Clean up CUDAWrappers::MatrixFree::reinit interface

dangars committed:
Remove destructor

dangars committed:
Exchange vectors in std::vector::swap()

masterleinad committed:
Deal with empty ranges in CUDAWrappers::MatrixFree::internal_reinit

masterleinad committed:
Address row_start

masterleinad committed:
Use integers in cudaMemset

masterleinad committed:
Add test

masterleinad committed:
Merge pull request #7965 from dangars/vector_swap

Exchange vectors of std::vector::swap() in step-18

masterleinad committed:
Merge pull request #7964 from dangars/destructor-step-40

Remove destructor in step40

luca-heltai committed:
Changed default parameter to true.

bangerth committed:
Merge pull request #7957 from rezarastak/DerivativeForm_doc_improvement

Mathematical notation of DerivativeForm

dangars committed:
Use the function n_locally_owned_active_cells()

drwells committed:
Merge pull request #7958 from bangerth/doc

Augment documentation in class IteratorRange.

drwells committed:
Merge pull request #7952 from bangerth/61-2

Remove pictures from the repository.

masterleinad committed:
Merge pull request #7972 from dangars/n_locally_owned_active_cells_step-18

Use the function n_locally_owned_active_cells() in step-18

masterleinad committed:
Merge pull request #7969 from masterleinad/use_integers_cuda_memset

Use integers in cudaMemset

jppelteret committed:
Fix some documentation of SD convenience functions

jppelteret committed:
Add functions to create SD substitution maps from scalar types

jppelteret committed:
Change function signature to use a template argument

jppelteret committed:
Add test for creation of SD substitution maps from scalar types

drwells committed:
Improve the documentation of parallel builds.

Not all users are familiar with the -j or --jobs flags for make and the
current explanation is buried in the middle of several other notes. We
can make this simpler by showing how to parallelize the build
explicitly.

masterleinad committed:
Merge pull request #7975 from drwells/improve-parallel-build-doc

Improve the documentation of parallel builds.

masterleinad committed:
Merge pull request #7961 from jppelteret/sd-map_utilities_01

SD scalar operations: Addition to substitution maps and map merging

masterleinad committed:
Export DEAL_II_PETSC_WITH_COMPLEX variable

masterleinad committed:
Improve doxygen documentation For ReadWriteVector

Rombur committed:
Merge pull request #7968 from masterleinad/empty_ranges_cuda_mf_internal_reinit

Deal with empty ranges in CUDAWrappers::MatrixFree::internal_reinit

bangerth committed:
Merge pull request #7977 from masterleinad/improve_docu_read_write

Improve doxygen documentation for ReadWriteVector

bangerth committed:
Update doc/news/changes/major/20190425RolandRichter

Co-Authored-By: anates <rolandar@stud.ntnu.no>

tjhei committed:
Merge pull request #7976 from masterleinad/export_petsc_with_complex

Export DEAL_II_PETSC_WITH_COMPLEX variable

masterleinad committed:
Merge pull request #7943 from anates/Relative_fe_degree

Degree of quadrature based on degree of the fe-variable

luzpaz committed:
Misc. typos

Found via `codespell -q 3 -I ../dealii-word-whitelist.txt -S ./bundled`

luzpaz committed:
Correction in previous commit per request

https://github.com/dealii/dealii/pull/7979#discussion_r279517855

drwells committed:
Merge pull request #7979 from luzpaz/misc-typos

Misc. typos

masterleinad committed:
Merge pull request #7967 from masterleinad/empty_cuda_kernels

Deal with empty empty sizes in CUDA kernels

masterleinad committed:
Restrict SymEngine test

jppelteret committed:
Merge pull request #7981 from masterleinad/restrict_symengine_test

Restrict SymEngine test

rezarastak committed:
Arpack Solver documentation fix

rezarastak committed:
newline bug in grid_in_abaqus fixed

rezarastak committed:
grid_in_abaqus test added

Rombur committed:
Merge pull request #7966 from masterleinad/cleanup_cuda_mf_reinit

Clean up CUDAWrappers::MatrixFree::reinit interface

Rombur committed:
Merge pull request #7982 from rezarastak/patch-1

Arpack Solver documentation fix

rezarastak committed:
fix typo in 2d_short_handwritten.inp

bangerth committed:
Use 'align' instead of 'eqnarray'.

rezarastak committed:
reverting the effects of the last commit on the last line of the inp file

jppelteret committed:
Build documentation with SymEngine enabled

jppelteret committed:
Add placeholder for documentation on symbolic differentiation

jppelteret committed:
Update module on user interfaces to AD libraries

Rombur committed:
Merge pull request #7983 from rezarastak/grid_in_abaqus_newline_fix

newline bug in grid_in_abaqus fixed

bangerth committed:
Merge pull request #7985 from jppelteret/ad-doc_02

Update module on user interfaces to AD libraries

drwells committed:
Merge pull request #7954 from bangerth/61-4

Edit the results section of step-61.

bangerth committed:
Merge pull request #7984 from jppelteret/sd-doc_01

Enable doxygen to build SD documentation

bangerth committed:
Align a set of equations.

bangerth committed:
Fix a couple of typos.

drwells committed:
Merge pull request #7987 from bangerth/fix-typos

Fix a couple of typos.

drwells committed:
Merge pull request #7986 from bangerth/61-5

Align a set of equations.

bangerth committed:
Merge pull request #7913 from luca-heltai/mapping-center

Mapping::get_center()

bangerth committed:
Merge pull request #7738 from tcclevenger/step-63

Step-63

masterleinad committed:
Merge pull request #7974 from jppelteret/sd-map_utilities_03

SD scalar operations: Creation of substitution maps

Commit:
step-61 announcement.

bangerth committed:
Minor update to a changelog entry.
